app/scripts/TrackRenderer.js

```diff
@@ -1033,13 +1032,44 @@ class TrackRenderer extends React.Component {
     return last;
   }
 
+  valueScaleZoom() {
+    const myWheelDelta = (dy, dm) => dy * (dm ? 120 : 1) / 500;
+
+    if (event.sourceEvent && event.sourceEvent.deltaY) {
+      // mouse move probably from a drag event
+      const dy = event.sourceEvent.deltaY;
+      const dm = event.sourceEvent.deltaMode;
+
+      const mwd = myWheelDelta(dy, dm);
+
+      const cp = clientPoint(this.props.canvasElement, event.sourceEvent);
+      for (const track of this.getTracksAtPosition(...cp)) {
+        track.zoomedY(cp[1] - track.position[1], 2 ** mwd);
+      }
+    } else if (event.sourceEvent && event.sourceEvent.movementY) {
+      // mouse wheel from zoom event
+      // const cp = clientPoint(this.props.canvasElement, event.sourceEvent);
+      for (const track of this.getTracksAtPosition(...this.zoomStartPos)) {
+        track.movedY(event.sourceEvent.movementY);
+      }
+    }
+
+    // reset the zoom transform
+    this.zoomTransform = this.zoomStartTransform;
+  }
+
   /**
    * Respond to a zoom event.
    *
    * We need to update our local record of the zoom transform and apply it
    * to all the tracks.
    */
   zoomed() {
+    if (this.props.valueScaleZoom || this.valueScaleZooming) {
```

Comment:
Would it make sense to directly check the `event` object here to see if `altKey` or `shiftKey` is `true`? That would simplify the code a little bit and avoid state changes.